Service parameter name `n` is converted to boolean

# Description:
I have created an IVCAP Service based on the [primitive](https://github.com/fogleman/primitive) command line tool, which has a parameter `n` to define the number of shapes that should be used to reproduce the input image. When attempting to define a service Parameter called `n` it gets converted to a boolean, resulting in the ivcap cli tool rejecting the defined service.

# Environment
ivcap-sdk-service==0.3.1

# Steps to Reproduce
Create a [service.py](examples/primitive/service.py) that contains an `ivcap_sdk_service.Service` object. An example of the relevant section is:

```Python
SERVICE = Service(
    description = 'Reproduce images with geometric primitives.',
    name = 'primitive',
    parameters = [
        .
        .        
        Parameter(            
            name = 'n',
            description = 'number of shapes',
            type = Type.INT,
            default = 100,
            optional = True
        ),
        .
        .
    ],
    workflow=PythonWorkflow(script='/primitive/service.py', min_memory='2Gi')
)
```

Print the service description via the command (making sure to use the appropriate provider and account identifier):
```bash
env IVCAP_PROVIDER_ID=urn:ivcap:provider:e91bb525-83b7-4641-9fb8-b61d084a9b81 \
    IVCAP_ACCOUNT_ID=urn:ivcap:account:e91bb525-83b7-4641-9fb8-b61d084a9b81 \ 
    IVCAP_CONTAINER=localhost:5000/ivcap.test/primitive_service:6fb5281 \
    python service.py --ivcap:print-service-description
```

The above command generates the following output (only the relevant output has been copied here):
```yaml
account-id: urn:ivcap:account:e91bb525-83b7-4641-9fb8-b61d084a9b81
description: Reproduce images with geometric primitives.
id: '@SERVICE_ID@'
name: primitive
parameters:
  .
  .
- default: '100'
  description: number of shapes
  name: n
  optional: true
  type: int
  .
  .

provider-id: urn:ivcap:provider:e91bb525-83b7-4641-9fb8-b61d084a9b81
workflow:
  basic:
    command:
    - python
    - /primitive/service.py
    image: localhost:5000/ivcap.test/primitive_service:6fb5281
    memory:
      request: 2Gi
  type: basic
```

So far, so good. Now try to create a service with this yaml:
```bash
env IVCAP_PROVIDER_ID=urn:ivcap:provider:e91bb525-83b7-4641-9fb8-b61d084a9b81 \
                IVCAP_ACCOUNT_ID=urn:ivcap:account:e91bb525-83b7-4641-9fb8-b61d084a9b81 \
                IVCAP_CONTAINER=localhost:5000/ivcap.test/primitive_service:97e52f2 \
        python service.py --ivcap:print-service-description \
        | ivcap service update --create ivcap:service:53d4033f-8e34-5f9e-9925-df5bb88134e4:primitive-service --format yaml -f - --timeout 600
```

Which yields the following output:
```bash
Error: json: cannot unmarshal bool into Go struct field ParameterDefT.parameters.name of type string
make: *** [service-register] Error 1
```

# Proposed Solution
So the first question: is the problem occurring a) during the generation of the yaml in the `ivcap_sdk_service`, or b) in the consumption of the yaml in the `ivcap` cli tool? 

**The short answer:** this can be fixed in the generation of yaml in the `ivcap_sdk_service` tool.

## IVCAP cli
IVCAP unmarshalls the YAML above via [ivcap-cli/pkg/adapter/payload.go](https://github.com/ivcap-works/ivcap-cli/blob/8cc073a2838f2e92dd0664d7d9b8c74865f8509e/pkg/adapter/payload.go#L68), converting the `name` from a value of `n` to `false` as seen below:
![image](https://github.com/ivcap-works/ivcap-service-sdk-python/assets/1073961/df2cca74-6254-4e1b-aefe-d5c9f8d943ce)

From what I understand of the yaml package we use, we can fix this issue in the ivcap cli tool by defining a struct that represents the Service yaml. But then we'd have to define structs for all yaml we send the `ivcap` cli...

Looking at this from another angle, if we put the value of name into single quotes, the problem disappears. 
```yaml
# This fails:
- default: '100'
  description: number of shapes
  name: n
  optional: true
  type: int

# This passes:
- default: '100'
  description: number of shapes
  name: "n"
  optional: true
  type: int
```

So if all string values are enclosed in double quotes, this may resolve the issue. 

### Proposed ivcap_sdk_service fix

YAML is generated in [src/ivcap_sdk_service/service.py](https://github.com/ivcap-works/ivcap-service-sdk-python/blob/1cb52d6ef9b97659f68baa283476593ed049ada3/src/ivcap_sdk_service/service.py#L165):

```python
def to_yaml(self) -> str:
    return yaml.dump(self.to_dict(), default_flow_style=False)
```

The proposed solution is to change this to:
```python
def to_yaml(self) -> str:
    as_dict = self.to_dict()
    as_yaml = yaml.dump(as_dict, default_flow_style=False, default_style='"')
    return as_yaml
```

The change sets the `default_style` parameter in `yaml.dump`, which controls how strings are formatted when they are converted to YAML format. In this case, the default_style parameter is set to '"', which means that all strings in the output YAML will be enclosed in double quotes. This is useful because it ensures that any special characters in the strings are properly escaped, which can prevent errors when the YAML is parsed.

The following yaml is produced when implementing this fix:
```yaml
"account-id": "urn:ivcap:account:e91bb525-83b7-4641-9fb8-b61d084a9b81"
"description": "Reproduce images with geometric primitives."
"id": "@SERVICE_ID@"
"name": "primitive"
"parameters":
- "description": "input file"
  "name": "input_image"
  "type": "artifact"
- "default": "100"
  "description": "number of shapes"
  "name": "n"
  "optional": !!bool "true"
  "type": "int"
- "default": "1"
  "description": "mode: 0=combo, 1=triangle, 2=rect, 3=ellipse, 4=cirle, 5=rotatedrect,\
    \ 6=beziers, 7=rotatedellipse, 8=polygon"
  "name": "m"
  "optional": !!bool "true"
  "type": "int"
- "default": "0"
  "description": "add N extra shapes each iteration with reduced search (mostly good\
    \ for beziers)"
  "name": "rep"
  "optional": !!bool "true"
  "type": "int"
- "default": "1"
  "description": "save every Nth frame (only when %d is in output path)"
  "name": "nth"
  "optional": !!bool "true"
  "type": "int"
- "default": "256"
  "description": "resize large input images to this size before processing"
  "name": "r"
  "optional": !!bool "true"
  "type": "int"
- "default": "1024"
  "description": "output image size"
  "name": "s"
  "optional": !!bool "true"
  "type": "int"
- "default": "128"
  "description": "color alpha (use 0 to let the algorithm choose alpha for each shape)"
  "name": "a"
  "optional": !!bool "true"
  "type": "int"
- "default": "avg"
  "description": "starting background color (hex)"
  "name": "bg"
  "optional": !!bool "true"
  "type": "string"
- "default": "0"
  "description": "number of parallel workers (default uses all cores)"
  "name": "j"
  "optional": !!bool "true"
  "type": "int"
"provider-id": "urn:ivcap:provider:e91bb525-83b7-4641-9fb8-b61d084a9b81"
"workflow":
  "basic":
    "command":
    - "python"
    - "/primitive/service.py"
    "image": "localhost:5000/ivcap.test/primitive_service:6fb5281"
    "memory":
      "request": "2Gi"
  "type": "basic"
```

Which is successfully consumed by the ivcap cli:
```json
{
  "account": {
    "id": "urn:ivcap:account:e91bb525-83b7-4641-9fb8-b61d084a9b81"
  },
  "description": "Reproduce images with geometric primitives.",
  "id": "urn:ivcap:service:53d4033f-8e34-5f9e-9925-df5bb88134e4",
  "links": {
    "self": "http://ivcap.minikube/1/services/53d4033f-8e34-5f9e-9925-df5bb88134e4"
  },
  "name": "primitive",
  "parameters": [
    {
      "description": "input file",
      "label": "",
      "name": "input_image",
      "optional": false,
      "type": "artifact",
      "unit": ""
    },
    {
      "default": "100",
      "description": "number of shapes",
      "label": "",
      "name": "n",
      "optional": true,
      "type": "int",
      "unit": ""
    },
    {
      "default": "1",
      "description": "mode: 0=combo, 1=triangle, 2=rect, 3=ellipse, 4=cirle, 5=rotatedrect, 6=beziers, 7=rotatedellipse, 8=polygon",
      "label": "",
      "name": "m",
      "optional": true,
      "type": "int",
      "unit": ""
    },
    {
      "default": "0",
      "description": "add N extra shapes each iteration with reduced search (mostly good for beziers)",
      "label": "",
      "name": "rep",
      "optional": true,
      "type": "int",
      "unit": ""
    },
    {
      "default": "1",
      "description": "save every Nth frame (only when %d is in output path)",
      "label": "",
      "name": "nth",
      "optional": true,
      "type": "int",
      "unit": ""
    },
    {
      "default": "256",
      "description": "resize large input images to this size before processing",
      "label": "",
      "name": "r",
      "optional": true,
      "type": "int",
      "unit": ""
    },
    {
      "default": "1024",
      "description": "output image size",
      "label": "",
      "name": "s",
      "optional": true,
      "type": "int",
      "unit": ""
    },
    {
      "default": "128",
      "description": "color alpha (use 0 to let the algorithm choose alpha for each shape)",
      "label": "",
      "name": "a",
      "optional": true,
      "type": "int",
      "unit": ""
    },
    {
      "default": "avg",
      "description": "starting background color (hex)",
      "label": "",
      "name": "bg",
      "optional": true,
      "type": "string",
      "unit": ""
    },
    {
      "default": "0",
      "description": "number of parallel workers (default uses all cores)",
      "label": "",
      "name": "j",
      "optional": true,
      "type": "int",
      "unit": ""
    }
  ],
  "provider": {
    "id": "urn:ivcap:provider:e91bb525-83b7-4641-9fb8-b61d084a9b81"
  }
}
```

--- a/src/ivcap_sdk_service/service.py
+++ b/src/ivcap_sdk_service/service.py
@@ -163,7 +163,9 @@ class Service(JSONWizard):
         return d
 
     def to_yaml(self) -> str:
-        return yaml.dump(self.to_dict(), default_flow_style=False)
+        as_dict = self.to_dict()
+        as_yaml = yaml.dump(as_dict, default_flow_style=False, default_style='"')
+        return as_yaml
 
     def append_arguments(self, ap: ArgumentParser) -> ArgumentParser:
         type2type = {

